src/common/com/intellij/plugins/haxe/ide/index/HaxeComponentFileNameIndex.java

```diff
@@ -99,17 +101,26 @@ public static boolean getFileNames(@NotNull String qName,
     @NotNull
     public Map<String, Void> map(final FileContent inputData) {
       final PsiFile psiFile = inputData.getPsiFile();
+      final String moduleName = FileUtil.getNameWithoutExtension(inputData.getFileName());
       final List<HaxeClass> classes = HaxeResolveUtil.findComponentDeclarations(psiFile);
       if (classes.isEmpty()) {
         return Collections.emptyMap();
```

Comment:
If file not declares types, I think we should not generate index for files, so just keep the check in these lines:
```if (classes.isEmpty()) {
          return Collections.emptyMap();```

For clarification what is haxe specification says:
All files is `Haxe-Files`. Haxe file can define types. So to import we should write:
`import a.b.Bar.Bar` - even if there is one file, and one class inside.
But Haxe provide shortcut if file name equals type name, so we can write shorter: 
`import a.b.Bar`

`import a.b.Bar; ` or `import a.b.Bar.Bar; ` is allowed by haxe compiler to import Bar class.
`import a.b.Bar.doSomething;` or `import a.b.Bar.Bar.doSomething;` is allowed for importing  static method to use it without access the base class